README.md

```diff
@@ -0,0 +1,117 @@
+
+##Description
+
+The idea is to use Kodein replacing part of our production code using test doubles during the instrumentation tests execution. This allow us to write instrumentation tests easily.
+
+For writing those tests we are going to use a property that Kodein give us, the ability to override dependencies in a graph.
+
+As Android does not give us the possibility to provide dependencies by constructor, we need to use Kodein to replace those dependencies. This is the main issue we fix using "override dependencies" Kodein feature.
+
+In this example we can find a simple Activity with two dependencies. The first dependency is just related to the Activity execution context. Having dependencies that only live during the activity lifecycle, like a Presenter, a controller, or any visual dependency is quite common. You can find it in ActivityScopeClass.kt. The second dependency is an Application scope dependency. This dependency lifecycle is linked to the Application lifecycle and not to the Activity lifecycle as the first dependency does. You can find this example in the class ApplicationScopeClass.kt. This class is defined as a Singleton just to show that we can override singletons provision using any type of test dobules.
+
+We wrote a simple test class using Espresso to check if the text shown to the user is the one obtained from the production code dependency or our test double.
+
+You can find it in ```MainActivityTest.kt```. If you check one of the tests you can discover how we are using Kodein to override the dependencies.
+
+```java
+@Mock private lateinit var applicationScopeClass: ApplicationScopeClass
+
+@Test
+    fun shouldShowInjectedNameProvidedByTheDomainWhenReplaceByAMock() {
+        whenever(applicationScopeClass.getText()).thenReturn("Mock Name")
+        startActivity()
+
+        onView(withText("Mock Name")).check(matches(isDisplayed()))
+    }
+```
+
+We are creating a  Mock (we are using Mockito to provide test doubles, this is not mandatory, you 
+could use your own test doubles, or using other mocking frameworks). We are replacing the real call
+ to getText() for a mock value called 'Mock Name". We launch the activity and we use Expresso to verify that the mocked test has been displayed on the screen.
+
+For this test works perfectly we need to use Kodein to some magic happens. If you continue reading the class at the end of the file you can find the Kodein provisioning dependencies.
+
+```java
+    val testDependencies = Module(allowSilentOverride = true) {
+        bind<ApplicationScopeClass>() with instance(applicationScopeClass)
+        bind<ActivityScopeClass>() with instance(activityScopeClass)
+    }
+```
+
+we create a Module that allows overriding the graph, and we create provisioning for those classes. Depends on your objective you can use instance or provision method. In our case, we used the instance because we are initing Mockito in the test environment setup.
```

Comment:
I'd rewrite this:

> we create a Module that allows overriding the graph, and we create provisioning for those classes. Depends on your objective you can use instance or provision method. In our case, we used the instance because we are initing Mockito in the test environment setup.

as:

> We have created a module that allows us to override the graph and create a different provisioning for those classes. Depends on your objective you can use instance or provision methods. In our case, we use the instance because we are initing Mockito in the test setup.

Reply:
checked